### PR TITLE
Resolve #1458 -- Ezreal E targeting nexus and inibhitors

### LINKED ISSUE #1458: Ezreal E targeting nexus and inibhitors.
Seems like Ezreal arcane shift missile is targeting more targets than supposed to.

Ezreal E

### PR DESCRIPTION
**Pull request by vhnunes**
Really simple fix.

### CODE REVIEW

**Comment by vhnunes** [comment]
Is there any better way to figure out if the target is a non "dynamic" unit? Like tower, inhibitor and nexus.

**Comment by lzardy** [comment]
> Is there any better way to figure out if the target is a non "dynamic" unit? Like tower, inhibitor and nexus.

You can utilize the `SpellSector` class which supports a `SpellFlags` option, defaulting to the spell's normal behavior.
The script was made before sectors were available for use.

**Comment by vhnunes** [comment]
> > Is there any better way to figure out if the target is a non "dynamic" unit? Like tower, inhibitor and nexus.
> 
> You can utilize the `SpellSector` class which supports a `SpellFlags` option, defaulting to the spell's normal behavior. The script was made before sectors were available for use.

Done! Made some changes to spell sector to get unit hits ordered by descending distance.

File: Content/LeagueSandbox-Scripts/Characters/Ezreal/E.cs [hunk 1]
```diff
@@ -82,6 +90,26 @@ public void OnSpellPostChannel(ISpell spell)
 
         public void OnUpdate(float diff)
         {
+            if (_owner == null || _sector == null)
```
Comment (lzardy) [inline Content/LeagueSandbox-Scripts/Characters/Ezreal/E.cs:94]: Move all of this into a `TargetExecute` function. Refer to [Annie W](https://github.com/LeagueSandbox/GameServer/blob/indev/Content/LeagueSandbox-Scripts/Characters/Annie/W.cs#L26) implementation.

Comment (vhnunes) [reply Content/LeagueSandbox-Scripts/Characters/Ezreal/E.cs:94]: 🔨 https://github.com/LeagueSandbox/GameServer/pull/1459/commits/8424234e6f2418da5caf45a5c57d7a4b7761e46d

**Review by lzardy — APPROVED** [review]
Good.

**Pull request merged**